src/VisualStudio/CSharp/Impl/Options/Formatting/CodeStylePage.cs

```diff
@@ -151,18 +151,18 @@ private static void GetCurrentEditorConfigOptionsCSharp(OptionSet optionSet, Str
 
         private static void CSharpFormattingOptions_GenerateEditorconfig(OptionSet optionSet, Option<bool> option, StringBuilder editorconfig)
         {
-            var element = (EditorConfigStorageLocation<bool>)option.StorageLocations.OfType<IEditorConfigStorageLocation>().SingleOrDefault();
+            var element = option.StorageLocations.OfType<EditorConfigStorageLocation<bool>>().FirstOrDefault();
             if (element != null)
             {
                 editorconfig.Append(element.KeyName + " = ");
 
-                editorconfig.AppendLine(optionSet.GetOption(CSharpFormattingOptions.NewLineForElse).ToString().ToLower());
+                editorconfig.AppendLine(optionSet.GetOption(option).ToString().ToLower());
```

Comment:
ah, i see.  this is why so many values changed.

This made me realize something as well... As part of a *followup* PR, we should likely add asserts that show that the output you produce "roundtrips" properly with the original Option value.  i.e. the code should look something akin to this:

```c#
var serializedOptionText = <your code>;
var deserializedOption = location.Parse(serializedOptionText);
Debug.Assert(option.Equals(deserializedOption);
```

This would help validate that we're actually producing roundtrippable options.  This would have hopefully caught the issue (and any others where we're not serializing the option properly).